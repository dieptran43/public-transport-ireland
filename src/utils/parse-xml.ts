import { Parser } from 'xml2js'

export function parseXml (xmlString : string) : Promise<any> {
  return new Promise((resolve, reject) => {
    const p = new Parser()
    p.parseString(xmlString, (err : Error, result : any) => {
      if (err) {
        return reject(err)
      }

      return resolve(result)
    })
  })
}
