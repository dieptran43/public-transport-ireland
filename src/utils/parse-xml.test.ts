import { parseXml } from './parse-xml'

test('it should parse some xml successfully', async function () {
  const xml = '<a><b>c</b></a>'
  const data = await parseXml(xml)

  expect(data).toMatchSnapshot()
})

test('it should throw an exception with invalid xml', async function () {
  const xml = '<a><b>c'

  await expect(parseXml(xml)).rejects.toMatchSnapshot()
})
