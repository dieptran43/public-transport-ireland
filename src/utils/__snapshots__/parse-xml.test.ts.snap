// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it should parse some xml successfully 1`] = `
Object {
  "a": Object {
    "b": Array [
      "c",
    ],
  },
}
`;

exports[`it should throw an exception with invalid xml 1`] = `
[Error: Unclosed root tag
Line: 0
Column: 7
Char: ]
`;
